modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheGroupMetricsImpl.java

```diff
@@ -126,13 +133,22 @@ public CacheGroupMetricsImpl(CacheGroupContext ctx) {
             this::getTotalAllocatedSize,
             "Total size of memory allocated for group, in bytes.");
 
-        mreg.register("StorageSize",
-            this::getStorageSize,
-            "Storage space allocated for group, in bytes.");
+        if (ctx.shared().database() instanceof GridCacheDatabaseSharedManager) {
+            mreg.register("StorageSize",
+                () -> database().forGroupPageStores(ctx, PageStore::size),
+                "Storage space allocated for group, in bytes.");
 
-        mreg.register("SparseStorageSize",
-            this::getSparseStorageSize,
-            "Storage space allocated for group adjusted for possible sparsity, in bytes.");
+            mreg.register("SparseStorageSize",
+                () -> database().forGroupPageStores(ctx, PageStore::getSparseSize),
+                "Storage space allocated for group adjusted for possible sparsity, in bytes.");
+
+            storageSize = mreg.findMetric("StorageSize");
+            sparseStorageSize = mreg.findMetric("SparseStorageSize");
+        }
+        else {
+            storageSize = new AtomicLongMetric("NO_OP", null);
```

Comment:
Why do we need this `NO_OP` metrics?

Reply:
@nizhikov This metric need to provide zero value when PDS disabled (we do not register those metrics in monitoring but it should be available from java API). For example, it can be read from deprecated CacheGroupMetricsMXBeanImpl.

Reply:
I think it can be reworked as `storageSize == null ? 0 : storageSize.longValue()`.
Let's do it?

Reply:
@nizhikov Done